sdk/cosmos/azure-cosmos-spark_3_2-12/src/test/scala/com/azure/cosmos/spark/TestUtils.scala

```diff
@@ -272,4 +272,16 @@ object Platform {
   def isWindows: Boolean = {
     System.getProperty("os.name").toLowerCase.contains("win")
   }
+
+  // Indicates whether a test is capable of running when it attempts to access DirectByteBuffer via reflection.
+  // Spark 3.1 was written in a way where it attempts to access DirectByteBuffer illegally via reflection in Java 16+.
+  def canRunTestAccessingDirectByteBuffer: (Boolean, Any) = {
```

Comment:
@FabianMeiswinkel I added this instead of a more complicated Maven change. In short, Spark 3.1 isn't good friends with Java 16+ due to changes in reflective access to JDK internals.